src/gwt/src/org/rstudio/studio/client/workbench/views/console/ConsolePane.java

```diff
@@ -161,6 +172,57 @@ public void execute()
             }
          });
       }
+      else if (profilerMode_)
+      {
+         loadProfilerToolsIntoSecondaryToolbar();
+      }
+   }
+
+   @Override
+   public void setProfilerMode(boolean profilerMode)
+   {
+      if (profilerMode == profilerMode_)
+      {
+         return;
+      }
+
+      profilerMode_ = profilerMode;
+      setSecondaryToolbarVisible(debugMode_ || profilerMode_);
```

Comment:
Nit: you could call the below code to set up the toolbar state based on `debugMode_` and `profilerMode_` from both mode switch functions